app/src/main/java/at_challenge/deaguiar/daniel/atchalleng/controller/RouteDetailActivity.java

```diff
@@ -0,0 +1,203 @@
+package at_challenge.deaguiar.daniel.atchalleng.controller;
+
+import android.app.ActionBar;
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Handler;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import org.json.JSONException;
+import java.util.ArrayList;
+
+import at_challenge.deaguiar.daniel.atchalleng.model.BusStopList;
+import at_challenge.deaguiar.daniel.atchalleng.service.DownloadIntentService;
+import at_challenge.deaguiar.daniel.atchalleng.service.DownloadResultReceiver;
+import at_challenge.deaguiar.daniel.atchalleng.model.BusStop;
+import at_challenge.deaguiar.daniel.atchalleng.model.Departure;
+import at_challenge.deaguiar.daniel.atchalleng.model.DepartureList;
+import at_challenge.deaguiar.daniel.atchalleng.R;
+
+/**
+ * RouteDetailsActivity
+ * Responsible to fetch route data from server and show it to the user
+ *
+ * @author Daniel Besen de Aguiar
+ */
+public class RouteDetailActivity extends Activity implements DownloadResultReceiver.Receiver {
+
+    public static final String EXTRA_ROUTE_ID   = "id";
```

Comment:
Why are you keeping these constants in the activity classes? Wouldn't be better to move them to a service class?